src/DataManagement/MonthlyReport/MonthlyReportCalculator.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using ePiggyWeb.DataBase;
+using ePiggyWeb.DataManagement.Entries;
+using ePiggyWeb.DataManagement.Goals;
+using ePiggyWeb.Utilities;
+
+namespace ePiggyWeb.DataManagement.MonthlyReport
+{
+    public class MonthlyReportCalculator
+    {
+        private int UserId { get; }
+        private GoalDatabase GoalDatabase { get; }
+        private EntryDatabase EntryDatabase { get; }
+
+        private IEntryList Expenses { get; set; }
+        private IEntryList Income { get; set; }
+        private decimal AllSavings { get; set; }
+        private IGoalList Goals { get; set; }
+
+        private MonthlyReportResult Result { get; set; }
+
+
+
+        public MonthlyReportCalculator(GoalDatabase goalDatabase, EntryDatabase entryDatabase, int userId)
+        {
+            GoalDatabase = goalDatabase;
+            EntryDatabase = entryDatabase;
+            UserId = userId;
+        }
+
+        public async Task<MonthlyReportResult> Calculate()
+        { 
+            Result = await GatherDataAsync();
+            CalculateBiggestExpensesCategory();
+            Result.SavedUpGoals = CalculateSavedUpGoals();
+            CalculateGoalReport();
+            return Result;
+        }
+
+        private async Task<MonthlyReportResult> GatherDataAsync()
+        {
+            var today = DateTime.Today;
+            var month = new DateTime(today.Year, today.Month, 1);
+            var first = month.AddMonths(-1);
+            var last = month.AddDays(-1);
+            var expenses = (await EntryDatabase.ReadListAsync(UserId, EntryType.Expense)).GetSum();
+            var income = (await EntryDatabase.ReadListAsync(UserId, EntryType.Income)).GetSum();
+            AllSavings = income - expenses;
+            Expenses = (await EntryDatabase.ReadListAsync(UserId, EntryType.Expense)).GetFrom(first).GetTo(last);
+            Income = (await EntryDatabase.ReadListAsync(UserId, EntryType.Income)).GetFrom(first).GetTo(last);
```

Comment:
I think it would be more efficient to store the expenses and income as variables, and then to call getsum or any filter method on that, in order to not call the database more than necessary, as in:

           var totalExpenses = (await EntryDatabase.ReadListAsync(UserId, EntryType.Expense));
           var totalIncome = (await EntryDatabase.ReadListAsync(UserId, EntryType.Income));
           AllSavings = totalIncome .GetSum() - totalExpenses .GetSum();
           ...